operator/v1/types_ingress.go

```diff
@@ -166,6 +166,12 @@ type IngressControllerSpec struct {
 	// +optional
 	TLSSecurityProfile *configv1.TLSSecurityProfile `json:"tlsSecurityProfile,omitempty"`
 
+	// clientTLS specifies settings for requesting and verifying client
+	// certificates, which can be used to enable mutual TLS.
+	//
+	// +optional
+	ClientTLS ClientTLS `json:"clientTLS,omitempty"`
```

Comment:
optional+omitempty for a non-pointer does not make sense.

Reply:
Could you elaborate? We've been using this pattern in recent API changes since I thought we were trying to move away from using pointers where possible. e.g. https://github.com/openshift/api/pull/868

Reply:
Structs and field serialize differently.  The struct will always be serialized.  If an empty struct can be treated the same as nil struct, for configuration-style APIs, having non-pointers is easier to consume and discover.

Reply:
I've deleted `omitempty`.